File: src/frontend/src/domains/entity/model/baseModel.js
import { type } from '@/domains/value/model/type'

/**
 * 基本モデルのエンティティ
 */
export class BaseModel {
  /**
   * コンストラクタ
   *
   * @param {Object} data モデルデータ
   */
  constructor(data) {
    /**
     * モデル名
     * @type {string}
     */
    this.id = data?.id

    /**
     * 表示名
     * @type {string}
     */
    this.screenName = data?.screenName

    /**
     * ソース
     * @type {string}
     */
    this.source = data?.source

    /**
     * モデルクラス
     * このクラスがbase-modelなので、現実的に固定値
     * @type {type}
     */
    this.type = new type(data?.type)

    /**
     * リビジョン
     * @type {array<string>}
     */
    this.revisions = []
    if (Array.isArray(data?.revisions)) {
      for (const revision of data.revisions) {
        this.revisions.push(revision)
      }
    }

    /**
     * 追加情報
     *
     * @todo いまのところ使っていないので型定義していないが、使う場合は型定義する
     * @type {Object}
     */
    this.appends = data?.appends
  }

  /**
   * 値を取得する
   *
   * @returns {Object}
   */
  getValues() {
    return {
      id: this.id,
      screenName: this.screenName,
      source: this.source,
      type: this.type.toString(),
      revisions: this.revisions,
      appends: this.appends
    }
  }
}

File: src/backend/yt_diffuser/domains/entity/model/base_model.py
"""
基本モデルのエンティティ
"""
from typing import List, Dict

from pydantic import BaseModel
from huggingface_hub.utils import CachedRepoInfo

from yt_diffuser.types.enum.model import ModelType, ModelSource

class BaseModelEntity(BaseModel):
    """
    ベースモデルのエンティティ
    """

    id:str = None
    """
    モデル名
    """

    screen_name:str = None
    """
    表示名
    """

    source:ModelSource = ModelSource.HF
    """
    ソース
    """

    type:ModelType = ModelType.BASE_MODEL
    """
    モデル種類
    """

    revisions:List[str] = []
    """
    リビジョン
    """

    appends:Dict = {}
    """
    追加情報

    @todo いまのところ使っていないので型定義していないが、使う場合は型定義する
    """
    
    def add_revision(self, revision:str):
        """
        リビジョンを追加する
        """
        self.revisions.append(revision)
    
    @classmethod
    def from_cached_repo_info(cls, repo_info:CachedRepoInfo) -> 'BaseModelEntity':
        """
        Huggingface_hubのCacheRepoInfoからモデル情報を生成する

        args:
            repo_info: CachedRepoInfo キャッシュリポジトリ情報

        """
        model = BaseModelEntity( 
            id=repo_info.repo_id,
            screen_name= repo_info.repo_id,
            source= ModelSource.HF.value,
            type= ModelType.BASE_MODEL.value,
            revisions= [],
            appends= {}
        )


        for revision in repo_info.revisions:
            if revision.refs is None or len(revision.refs) == 0:
                model.add_revision(revision.commit_hash)
            else:
                for ref in revision.refs:
                    model.add_revision(ref)
        
        return model